db/operations.go

```diff
@@ -95,25 +95,33 @@ func (o *Operation) query() (string, error) {
 			updates[i] = fmt.Sprintf("%s=%s", columns[i], values[i])
 		}
 
-		return fmt.Sprintf("UPDATE %s SET %s WHERE %s = %s",
+		primaryKeySelector := getPrimaryKeyWhereClause(o.primaryKey)
+		return fmt.Sprintf("UPDATE %s SET %s WHERE %s",
 			o.table.identifier,
 			strings.Join(updates, ", "),
-			o.table.primaryColumn.escapedName,
-			escapeStringValue(o.primaryKey),
+			primaryKeySelector,
 		), nil
 
 	case OperationTypeDelete:
-		return fmt.Sprintf("DELETE FROM %s WHERE %s = %s",
+		primaryKeyWhereClause := getPrimaryKeyWhereClause(o.primaryKey)
+		return fmt.Sprintf("DELETE FROM %s WHERE %s",
 			o.table.identifier,
-			o.table.primaryColumn.escapedName,
-			escapeStringValue(o.primaryKey),
+			primaryKeyWhereClause,
 		), nil
 
 	default:
 		panic(fmt.Errorf("unknown operation type %q", o.opType))
 	}
 }
 
+func getPrimaryKeyWhereClause(primaryKey map[string]string) string {
+	reg := []string{}
```

Comment:
Let's add a case for `len(primaryKey) == 1` to avoid any allocation in this case, and let's pre-allocate the array when there is N elements:

```suggestion
    if len(primaryKey) == 1 {
        // Only way in Go to get first element of a map
        for key, value := range primaryKey {
            return key + " = " + value
        }
    }

	reg := make([]string, 0, len(primaryKey))
```

Reply:
Done

Reply:
I've also added escaping to the value. Done.